protocols/gossipsub/src/behaviour.rs

```diff
@@ -675,7 +676,7 @@ where
             } else {
                 // Explicit peers
                 for peer in &self.explicit_peers {
-                    if set.contains(peer) {
+                    if set.contains(&peer) {
```

Comment:
This is the part that if we are not flood_publishing, then now we send messages to all peers regardless if they are subscribed or not. 

I think this is not what we want to do and is not what the spec is implying.

Reply:
Wouldn't this send the message only to **explicit peers** found in the `set`?...regardless if they are subscribed or not, i.e. regardless how we calculate the `set` we'd forward to them only if they are set by the user as explicit peers....?

Reply:
...unless explicit peers are yet required to be subscribed to the topic and only their score is ignored..., is that what explicit peers are meant to be? if so, then yes, I see this is wrong.
I thought they were explicit and not even required to be subscribed to the topic.

Reply:
Ah yep, I didn't look too closely at this. You're right. Its the explicit peers. We still want to segregate them based on topic tho (I gave some reasons in the issue that links to this PR) :)